system/security/digisign/digisign.cpp

```diff
@@ -0,0 +1,273 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+#include "jliball.hpp"
+#ifdef _USE_OPENSSL
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#endif
+#include "jencrypt.hpp"
+#include "digisign.hpp"
+
+static CriticalSection digiSignCrit;
+static CriticalSection digiVerifyCrit;
+
+#define EVP_CLEANUP(key,ctx) EVP_PKEY_free(key);       \
+                             EVP_MD_CTX_destroy(ctx);
+
+#define EVP_THROW(str) {                                            \
+                         char buff[120];                            \
+                         ERR_error_string(ERR_get_error(), buff);   \
+                         throw MakeStringException(-1, str, buff);  \
+                       }
+
+
+class CDigitalSignatureManager : implements IDigitalSignatureManager
+{
+private:
+    StringAttr   publicKeyFile;
+    StringBuffer publicKeyBuff;
+    StringAttr   privateKeyFile;
+    StringBuffer privateKeyBuff;
+    StringBuffer passphraseBuffEnc;
+    bool         signingConfigured;
+    bool         verifyingConfigured;
+
+public:
+    CDigitalSignatureManager()
+        : signingConfigured(false), verifyingConfigured(false)
+    {
+#ifdef _USE_OPENSSL
+        //query private key file location from environment.conf
+        const char * pubKey, * privKey, * passPhrase;
+        bool rc = queryHPCCPKIKeyFiles(nullptr, &pubKey, &privKey, &passPhrase);
```

Comment:
@RussWhitehead 
This doesn't compile when I try it locally.  It is defined as 
```
extern jlib_decl bool queryHPCCPKIKeyFiles(const char * *  _certificate,//HPCCCertFile
                                           const char * *  _privateKey, //HPCCPrivateKeyFile
                                           const char * *  _passPhrase);//HPCCPassPhrase
```
And I don't see a change in this PR to add an extra parameter.

Reply:
Try refreshing your master, the new argument was added via https://github.com/hpcc-systems/HPCC-Platform/pull/10952